addons/utm/models/utm.py

```diff
@@ -42,14 +49,68 @@ def _group_expand_stage_ids(self, stages, domain, order):
         stage_ids = stages._search([], order=order, access_rights_uid=SUPERUSER_ID)
         return stages.browse(stage_ids)
 
+    def _merge_utm_campaigns(self, keep_campaign, merge_values={}):
+        if len(self.ids) <= 1:
+            raise UserError(_('Please select more than one campaign from the list.'))
+        if not all(campaign.active for campaign in self):
+            raise UserError(_('Only active campaigns can be merged.'))
+        if keep_campaign and keep_campaign not in self:
+            raise UserError(_('Kept campaign should be one of the merged campaign.'))
+
+        default_merge_values = {
+            'tag_ids': reduce(lambda tag_ids_a, tag_ids_b: tag_ids_a | tag_ids_b, (campaign.tag_ids for campaign in self)),
+            'is_website': any(campaign.is_website for campaign in self)
+        }
+        merge_values.update(default_merge_values)
+        keep_campaign.write(merge_values)
+
+        # deactive uneeded campaigns, set reference to the merged campaign, and update reference of other campaigns if the reference is deactived
+        deactived_campaigns = self - keep_campaign
+        deactived_campaigns.write({'active': False, 'reference_utm_campaign_id': keep_campaign.id})
+        self.search([('reference_utm_campaign_id', 'in', deactived_campaigns.ids)]).write({
+            'reference_utm_campaign_id': keep_campaign.id
+        })
+
+        self._redirect_all_to_merged_campaign(keep_campaign, deactived_campaigns)
+
+        return keep_campaign, deactived_campaigns
+
+    def _redirect_all_to_merged_campaign(self, merged_campaign, deactived_campaigns):
```

Comment:
Should be called on a recordset instead of making it an api.model.

`merged_campaigns._redirect_all_to_merged_campaign(keep_campaign)`

Also rename the method to `_clean_merged_campaigns_references`.
(And `_clean_merged_campaigns_reference` for the other one).